core/test_network/src/lib.rs

```diff
@@ -818,8 +818,8 @@ impl TestClient for Client {
     fn test_with_key(api_url: &SocketAddr, telemetry_url: &SocketAddr, keys: KeyPair) -> Self {
         let mut configuration = ClientConfiguration::test(api_url, telemetry_url);
         let (public_key, private_key) = keys.into();
-        configuration.public_key = public_key;
-        configuration.private_key = private_key;
+        configuration.set_public_key(public_key);
```

Comment:
Consider `set_key_pair((public_key, private_key))`. It saves one line, but in a lot of places.